src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java

```diff
@@ -103,27 +99,26 @@ public ApplicationDcContext(Context context) {
     }
     // /migration
 
-    try {
-      PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
-
-      imapWakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "imapWakeLock");
-      imapWakeLock.setReferenceCounted(false); // if the idle-thread is killed for any reasons, it is better not to rely on reference counting
-
-      mvboxWakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "mvboxWakeLock");
-      mvboxWakeLock.setReferenceCounted(false); // if the idle-thread is killed for any reasons, it is better not to rely on reference counting
-
-      sentboxWakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "sentboxWakeLock");
-      sentboxWakeLock.setReferenceCounted(false); // if the idle-thread is killed for any reasons, it is better not to rely on reference counting
+    new Thread(() -> {
+      DcEventEmitter emitter = getEventEmitter();
+      while (true) {
+        DcEvent event = emitter.getNextEvent();
+        if (event==null) {
```

Comment:
I don't understand how this condition can be met. 
getNextEvent returns a `DcEvent(getNextEventCPtr())` and this is cannot be null, or?

Reply:
it is null if the context is unref'd, this is the new gist of graceful shutdowns

Reply:
sorry, you're right. there is an error in the bindings.

Reply:
fixed that, good catch! thanks!